### PR TITLE
cloudflare: allow user configuration of results per page

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

See title.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

-

### CODE REVIEW

File: packages/cloudflare/data_stream/audit/agent/stream/httpjson.yml.hbs [hunk 1]
```diff
@@ -5,7 +5,11 @@ request.tracer.filename: "../../logs/httpjson/http-request-trace-*.ndjson"
 request.tracer.maxbackups: 5
 {{/if}}
 request.method: "GET"
+{{#if per_page}}
+request.url: {{api_url}}/client/v4/accounts/{{account}}/audit_logs?page=1&per_page={{per_page}}&direction=desc
```
Comment (chrisberkhout) [inline packages/cloudflare/data_stream/audit/agent/stream/httpjson.yml.hbs:9]: Looks good, matches API doc https://developers.cloudflare.com/api/operations/audit-logs-get-account-audit-logs

File: packages/cloudflare/data_stream/logpull/agent/stream/httpjson.yml.hbs [hunk 2]
```diff
@@ -28,6 +28,11 @@ request.transforms:
   - set:
       target: header.X-Auth-Key
       value: {{escape_string auth_key}}
+{{/if}}
+{{#if per_page}}
+  - set:
+      target: url.params.per_page
```
Comment (chrisberkhout) [inline packages/cloudflare/data_stream/logpull/agent/stream/httpjson.yml.hbs:34]: For this endpoint, it looks like it should be `count`, not `per_page`.
https://developers.cloudflare.com/api/operations/logs-received-get-logs-received

**Review by chrisberkhout — CHANGES_REQUESTED** [review]
Aside: the API docs mention "Logpush", but not "Logpull". However, the `/client/v4/zones/{zone_identifier}/logs/received` endpoint is documented under "Logs Received -> Get logs received".

File: packages/cloudflare/data_stream/logpull/agent/stream/httpjson.yml.hbs (hunk 2 above)
Comment (efd6) [reply packages/cloudflare/data_stream/logpull/agent/stream/httpjson.yml.hbs:34]: Changing the param target, but leaving the rest as `per_page` for consistency with the other data stream.

**Review by chrisberkhout — APPROVED** [review]

**Pull request merged**